### ISSUE TITLE
Construct DAGs from YAML

### ISSUE THREAD

**Issue by chriscardillo** [apache/airflow#10009]
**Description**

The ability to construct a DAG, that DAG's jobs, and the dependencies between those jobs (and external dependencies) can be time-consuming and cumbersome if you haven't figured out a way to dynamically create your DAGs. Across the internet, there exist packages and blog posts on how one might dynamically construct their DAGs.

I think it would be valuable to newcomers and Airflow veterans alike if Airflow contained an option to dynamically set up DAGs using YAML. While I have built out a package, [gusty](https://github.com/chriscardillo/gusty), that does a pretty good job of dynamically generating DAGs, I think the wider Airflow community would benefit from a feature like this existing directly in Airflow. (There is also a fully dockerized [demo](https://github.com/chriscardillo/gusty-demo) of this feature, which I will discuss more below.)

**Use case / motivation**

Configuring a DAG using YAML, where each .yml file is a specific job, would allow for an ease in the management of DAGs, as well as a simpler way to add/adjust/delete jobs as needed. While, in my opinion, one of the beautiful parts of Airflow is how you _can_ configure it however you'd like, there are some people (and I used to be one of them) who could really use a little guidance in what _might_ be a good approach to building out a good Airflow instance.

You can check out a fully containerized example of a tool that turns YAML into DAGs [here](https://github.com/chriscardillo/gusty-demo). I hope the demo and the readme do a good job of illustrating the benefits of offering this approach within Airflow.

In short, I think a feature like this, to construct DAGs from YAML, would help make Airflow more accessible to a wider audience by offering an option on how to construct DAGs and manage jobs/custom operators. I could also be completely wrong.

Thanks for taking the time out to read this and I hope to hear from you soon.

**Related Issues**

To the best of my knowledge there are not other issues relating to this.

**Comment by mik-laj** [comment]
Hello,

We are trying to focus on the development of support DAG as Python code, but this allows the use of other formats according to the needs of a specific organization.

The community has also developed support for other formats
https://github.com/ajbosco/dag-factory
https://github.com/Barski-lab/cwl-airflow

Can you add your plluguins to awesome listt? This will make it easier for us to find this plugin in the future.
https://github.com/jghoman/awesome-apache-airflow

**Comment by chriscardillo** [comment]
Thanks so much. I've opened a PR on the awesome list.
